src/communication/Subscriber.cpp

```diff
@@ -10,3 +10,30 @@
 // SPDX-License-Identifier: Apache-2.0
 
 #include "up-cpp/communication/Subscriber.h"
+
+namespace uprotocol::communication {
+
+[[nodiscard]] utils::Expected<std::unique_ptr<Subscriber>,
+                              uprotocol::v1::UStatus>
+Subscriber::subscribe(std::shared_ptr<transport::UTransport> transport,
+                      const v1::UUri& topic,
+                      transport::UTransport::ListenCallback&& callback) {
+	if (!transport) {
+		throw std::invalid_argument("transport cannot be null");
+	}
```

Comment:
Nitpick (ish) - The function definition in the header specifically says it returns either a unique pointer, or a UStatus error code -- so should this be a return instead of a throw?!?!?

Reply:
Well... I actually forgot to consider what the error handling for a null transport pointer would be when I wrote the header comments :sweat_smile: 

In keeping with the theme elsewhere, throwing an exception is probably the right choice here. A null transport would be indicative of a programmatic error and not some temporary condition.

Reply:
I wonder if we should add a `NullTransport` exception (not for this PR, but in the future)